index.js

```diff
@@ -13,17 +15,21 @@ module.exports = function (pid, signal, callback) {
         exec('taskkill /pid ' + pid + ' /T /F', callback);
         break;
     case 'darwin':
-        buildProcessTreeDarwin(pid, tree, pidsToProcess, function () {
-            killAll(tree, signal, callback);
-        });
-        break;
-    case 'sunos':
-        buildProcessTreeSunOS(pid, tree, pidsToProcess, function () {
+        buildProcessTree(pid, tree, pidsToProcess, function (parentPid) {
+          return spawn('pgrep', ['-P', parentPid]);
+        }, function () {
             killAll(tree, signal, callback);
         });
         break;
+    // case 'sunos':
+    //     buildProcessTreeSunOS(pid, tree, pidsToProcess, function () {
+    //         killAll(tree, signal, callback);
+    //     });
+    //     break;
```

Comment:
I don't have a `sunos` to test, but it should be possible to port the `ps` spawner from #6.